lib/swm050/gpio.c

```diff
@@ -0,0 +1,190 @@
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_toggle(uint16_t gpios)
+{
+	uint32_t curr_status = GPIO_DATA & gpios;
+	GPIO_DATA = (GPIO_DATA & (~gpios)) | (~curr_status);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Set the direction of a Group of Pins to Input
+
+Set the direction of one or more pins of GPIO to input.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_input(uint16_t gpios)
+{
+	GPIO_DIR &= ~gpios;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Set the direction of a Group of Pins to Output
+
+Set the direction of one or more pins of GPIO to output.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_output(uint16_t gpios)
+{
+	GPIO_DIR |= gpios;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Select the alternative function of a Group of Pins
+
+Select the alternative function of one or more pins of GPIO.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+@param[in] af_en Bool. Whether alternative function is selected
+*/
+void gpio_sel_af(uint16_t gpios, bool af_en)
+{
+	if (gpios & GPIO0) {
+		GPIO_SEL = (GPIO_SEL & (~0x3)) | (af_en ? 0x1 : 0x0);
+	}
+	if (gpios & GPIO1) {
+		GPIO_SEL = (GPIO_SEL & (~0xc)) | (af_en ? 0x4 : 0x0);
+	}
+	if (gpios & GPIO2) {
+		GPIO_SEL = (GPIO_SEL & (~0x30)) | (af_en ? 0x10 : 0x0);
+	}
+	if (gpios & GPIO7) {
+		GPIO_SEL = (GPIO_SEL & (~0xc000)) | (af_en ? 0x4000 : 0x0);
+	}
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Enable the internal pull-up of a Group of Pins
+
+Enable or disable the internal pull-up of one or more pins of GPIO.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+@param[in] en Bool. Whether pull-up is enabled
+*/
+void gpio_pullup(uint16_t gpios, bool en)
+{
+	if (en) {
+		GPIO_PULLUP |= gpios;
+	} else {
+		GPIO_PULLUP &= ~gpios;
+	}
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Enable the input function of a Group of Pins
+
+Enable or disable the input function of one or more pins of GPIO. Disabling
+the input function of pins can decrease the power usage of the MCU.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+@param[in] en Bool. Whether input function is enabled
+*/
+void gpio_in_en(uint16_t gpios, bool en)
```

Comment:
I'm afraid it's still not clear to me what the difference is between gpio_input and gpio_in_en.  I can _see_ thaht they're different registers, but I don't understand from the documentation when I need each one.  Particularly this one.